site/content/contribute/more-info/desktop/architecture/configuration.md

```diff
@@ -0,0 +1,49 @@
+---
+title: "Configuration"
+heading: "Configuration and server management"
+description: "Outlines the renderer processes that represent external Mattermost servers, and the modules that manage them."
+date: 2023-04-03T00:00:00-05:00
+weight: 1
+aliases:
+  - /contribute/desktop/architecture/configuration
+---
+
+### Config module
+
+The **configuration** module in the **Common** module is responsible for facilitating reading from and writing to external configuration sources. It also consolidates, verifies, and upgrades configuration where applicable.
+
+#### Files
+
+We have a few different configuration files in the Desktop App, but the main one is `config.json`. Most of the user's configuration from the Settings Window is stored there, as well as any user-configured servers.
+
+The application supports different configuration versions, and allows for them to be migrating to the version supported by the configuration module via the `upgradePreferences` module. When no configuration is found, the `defaultPreferences` object is copied over to the main configuration module.
+
+We also support a build configuration, in which the packager of the application can pre-define servers and a few other configuration items.
```

Comment:
I had to google this one since I don't think I've ever actually written a sentence containing "in which" before even though I know I've said it out loud before
```suggestion
We also support a build configuration in which the packager of the application can pre-define servers and a few other configuration items.
```